src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -220,6 +223,49 @@ else if (blockState.getBlock() == Blocks.BAMBOO_SAPLING)
         }
     }
 
+    /**
+     * For structure placement, check if two blocks are alike or if action has to be taken.
+     * @param blockState1 the first blockState.
+     * @param blockState2 the second blockState.
+     * @return true if nothing has to be done.
+     */
+    public static boolean areBlockStatesEqual(final BlockState blockState1, final BlockState blockState2)
+    {
+        if (blockState2 == null)
+        {
+            return true;
+        }
+
+        final Block structureBlock = blockState2.getBlock();
+        final Block worldBlock = blockState1.getBlock();
+
+        if (structureBlock == ModBlocks.blockSubstitution || blockState2.equals(blockState1))
+        {
+            return true;
+        }
+
+        if (structureBlock == ModBlocks.blockSolidSubstitution)
+        {
```

Comment:
wouldnt this sth we can move into the handler? instead of checking it before

Reply:
Then we have to handle all of them, and we don't want to have to handle them.